src/java.security.jgss/share/classes/sun/security/krb5/internal/crypto/EType.java

```diff
@@ -67,7 +96,19 @@ public static void initStatic() {
                                     exc.getMessage());
             }
         }
-        allowWeakCrypto = allowed;
+        if (allowWeakCrypto) {
+            result[num++] = EncryptedData.ETYPE_DES3_CBC_HMAC_SHA1_KD;
+            result[num++] = EncryptedData.ETYPE_ARCFOUR_HMAC;
+            result[num++] = EncryptedData.ETYPE_DES_CBC_CRC;
+            result[num++] = EncryptedData.ETYPE_DES_CBC_MD5;
+        }
+
+        // Weak crypto are also supported and can be enabled manually
+        if (num == result.length) {
+            supportedETypes = result;
+        } else {
+            supportedETypes = Arrays.copyOf(result, num);
+        }
```

Comment:
nit: it seems clearer to base supportedETypes on defaultETypes, e.g.
`        if (allowWeakCrypto) {
            supportedETypes = Arrays.copyOf(defaultETypes,
                defaultETypes.length + 4);
            supportedETypes[defaultETypes.length] =
                    EncryptedData.ETYPE_DES3_CBC_HMAC_SHA1_KD;
            supportedETypes[defaultETypes.length + 1] =
                    EncryptedData.ETYPE_ARCFOUR_HMAC;
            supportedETypes[defaultETypes.length + 2] =
                    EncryptedData.ETYPE_DES_CBC_CRC;
            supportedETypes[defaultETypes.length + 3] =
                    EncryptedData.ETYPE_DES_CBC_MD5;
        } else {
            supportedETypes = defaultETypes;
        }
`

Reply:
Yes.